### PR TITLE
Fix build warnings

### PR DESCRIPTION
**Pull request by alexeylysenko**
Fixed the warning messages appeared during build except following:
1) Q_DECL_DEPRECATED_X("Use uuid") QString id() const;
    Not sure whether I can simply replace id() to uuid().
2) Qt deprecation warnings
    If there is a plan to move to Qt6 it would be very useful to fix all warnings in Qt 5.15 to make a migration smoother - can provide a separate PR for that.

### CODE REVIEW

**Comment by TheOneRing** [comment]
At the moment we still rely on Qt 5.12 so...

**Comment by TheOneRing** [comment]
But as we don't plan to quit business we certainly plan to switch to Qt6 some day

File: test/testcredentialmanager.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ private Q_SLOTS:
                 });
             });
         });
-        QTest::qWaitFor([this] { return _finished; });
+        QVERIFY(QTest::qWaitFor([this] { return _finished; }));
```
Comment (alexeylysenko) [inline test/testcredentialmanager.cpp:84]: @TheOneRing could you please comment on that? I expect the test execution has to wait until job will be finished, right? Or I missed something...

File: src/gui/generalsettings.cpp [hunk 2]
```diff
@@ -302,15 +304,15 @@ void GeneralSettings::slotToggleOptionalDesktopNotifications(bool enable)
     cfgFile.setOptionalDesktopNotifications(enable);
 }
 
+#ifdef Q_OS_WIN
 void GeneralSettings::slotShowInExplorerNavigationPane(bool checked)
```
Comment (TheOneRing) [inline src/gui/generalsettings.cpp:308]: and in the header?

File: test/testcredentialmanager.cpp (hunk 1 above)
Comment (TheOneRing) [reply test/testcredentialmanager.cpp:84]: Yes the idea is to wait for the test to finish.
As we have the skip condition on linux the wait might time out here.

**Comment by TheOneRing** [comment]
> 
> 
> Fixed the warning messages appeared during build except following:
> 
>     1. Q_DECL_DEPRECATED_X("Use uuid") QString id() const;
>        Not sure whether I can simply replace id() to uuid().
> 

Thats why some are still around :)
>     2. Qt deprecation warnings
>        If there is a plan to move to Qt6 it would be very useful to fix all warnings in Qt 5.15 to make a migration smoother - can provide a separate PR for that.

**Pull request merged**